pkg/server/api/entry/v1/service_test.go

```diff
@@ -799,6 +799,448 @@ func setupServiceTest(t *testing.T, ds datastore.DataStore) *serviceTest {
+					ParentId: &types.SPIFFEID{TrustDomain: "", Path: "/invalid"},
+				},
+			},
+			dsError: errors.New("some error"),
+			expectLogs: []spiretest.LogEntry{
+				{
+					Level:   logrus.ErrorLevel,
+					Message: "failed to fetch entry",
+					Data: logrus.Fields{
+						logrus.ErrorKey: "some error",
+					},
+				},
+			},
+			expectResults: []*entrypb.BatchUpdateEntryResponse_Result{
+				{
+					Status: &types.Status{Code: int32(codes.Internal), Message: "failed to fetch entry"},
+				},
+			},
+		},
+		{
+			name:           "Check Output Masks Work",
+			initialEntries: []*types.Entry{initialEntry},
+			inputMask:      &types.EntryMask{
+				// With this empty, the update operation should be a no-op
+			},
+			outputMask: &types.EntryMask{
+				SpiffeId: true,
+			},
+			updateEntries: []*types.Entry{
+				{
+					Ttl: 500000,
+				},
+			},
+			expectResults: []*entrypb.BatchUpdateEntryResponse_Result{
+				{
+					Status: &types.Status{Code: int32(codes.OK), Message: "ok"},
+					Entry: &types.Entry{
+						SpiffeId: &types.SPIFFEID{TrustDomain: "example.org", Path: "/workload"},
+					},
+				},
+			},
+		},
+	} {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			test.logHook.Reset()
+
+			// First create the initial entries
+			createResp, err := test.client.BatchCreateEntry(ctx, &entrypb.BatchCreateEntryRequest{
+				Entries:    tt.initialEntries,
+				OutputMask: tt.outputMask,
+			})
+			require.NoError(t, err)
+			require.Equal(t, len(createResp.Results), len(tt.updateEntries))
+			// Be sure to clean up the entries created
+			defer func() {
+				ds.SetError(nil)
+				for i := range createResp.Results {
+					_, err := ds.DeleteRegistrationEntry(context.Background(), &datastore.DeleteRegistrationEntryRequest{EntryId: createResp.Results[i].Entry.Id})
+					require.NoError(t, err)
+				}
+			}()
+			// Then copy the IDs of the created entries onto the entries to be updated
+			updateEntries := tt.updateEntries
+			for i := range createResp.Results {
+				updateEntries[i].Id = createResp.Results[i].Entry.Id
+			}
+
+			ds.SetError(tt.dsError)
+
+			// Actually do the update, with the proper IDs
+			resp, err := test.client.BatchUpdateEntry(ctx, &entrypb.BatchUpdateEntryRequest{
+				Entries:    updateEntries,
+				InputMask:  tt.inputMask,
+				OutputMask: tt.outputMask,
+			})
+
+			spiretest.AssertLogs(t, test.logHook.AllEntries(), tt.expectLogs)
+			if tt.err != "" {
+				spiretest.RequireGRPCStatusContains(t, err, tt.code, tt.err)
+				require.Nil(t, resp)
+				return
+			}
+			require.Equal(t, len(tt.updateEntries), len(resp.Results))
+			require.NotNil(t, resp)
+
+			// The updated entries contain IDs, which we don't know before running the test.
+			// To make things easy we set all the IDs to empty before checking the contents.
+			for i := range resp.Results {
+				if resp.Results[i].Entry != nil {
+					resp.Results[i].Entry.Id = ""
+				}
+			}
+
+			spiretest.AssertProtoEqual(t, &entrypb.BatchUpdateEntryResponse{
```

Comment:
can you verify entries on DS? just to be sure that we send there waht we want.

Reply:
Working on this...

Reply:
Done